src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -317,6 +318,9 @@ public void waitForReady() throws InterruptedException {
                     LOGGER.log(Level.INFO,
                             "Master isn't ready to talk to us on {0}. Will retry again: response code={1}",
                             new Object[]{url, con.getResponseCode()});
+                } catch (SocketTimeoutException e) {
+                    LOGGER.log(INFO, "Failed to connect to the master. Will retry again: {0} {1}",
```

Comment:
I would not print "Failed to connect to the master" then. Maybe just "Socket Timeout while connectiong to the master"?

Reply:
I think this can be extended to `catch (SocketTimeoutException|ConnectException|NoRouteToHostException e)`

Reply:
fine for me